window/src/os/x11/window.rs

```diff
@@ -134,24 +136,24 @@ impl Drop for XWindowInner {
     }
 }
 
-unsafe impl HasRawDisplayHandle for XWindowInner {
-    fn raw_display_handle(&self) -> RawDisplayHandle {
-        let mut handle = XcbDisplayHandle::empty();
+impl HasDisplayHandle for XWindowInner {
+    fn display_handle(&self) -> Result<DisplayHandle<'_>, HandleError> {
         if let Some(conn) = self.conn.upgrade() {
-            handle.connection = conn.conn.get_raw_conn() as _;
-            handle.screen = conn.screen_num;
+            let handle =
+                XcbDisplayHandle::new(NonNull::new(conn.conn.get_raw_conn() as _), conn.screen_num);
+            unsafe { Ok(DisplayHandle::borrow_raw(RawDisplayHandle::Xcb(handle))) }
+        } else {
+            Err(HandleError::Unavailable)
         }
-
-        RawDisplayHandle::Xcb(handle)
     }
 }
 
-unsafe impl HasRawWindowHandle for XWindowInner {
-    fn raw_window_handle(&self) -> RawWindowHandle {
-        let mut handle = XcbWindowHandle::empty();
-        handle.window = self.child_id.resource_id();
-        handle.visual_id = self.conn.upgrade().unwrap().visual.visual_id();
-        RawWindowHandle::Xcb(handle)
+impl HasWindowHandle for XWindowInner {
+    fn window_handle(&self) -> Result<WindowHandle<'_>, HandleError> {
+        let mut handle =
+            XcbWindowHandle::new(NonZeroU32::new(self.window_id.resource_id()).expect("non-zero"));
```

Comment:
This resolves my blank window issue:

```suggestion
            XcbWindowHandle::new(NonZeroU32::new(self.child_id.resource_id()).expect("non-zero"));
```